### PR TITLE
fix warnings with VS

### PR DESCRIPTION
**Pull request by nunoplopes**
It's now /W2 clean.

We can't use \u with VS because the encoding is dependent on the locale. With the \xXX encoding we ensure it's UTF-8.

### CODE REVIEW

**Comment by Kha** [comment]
It's not much of a difference, but can't we use `u8"..."` literals?

**Comment by nunoplopes** [comment]
> It's not much of a difference, but can't we use u8"..." literals?

In VS, yes. I thought it was not used due to some compatibility issue with older compilers?

File: src/frontends/lean/pp.cpp [hunk 1]
```diff
@@ -127,13 +127,13 @@ void initialize_pp() {
     g_ellipsis_n_fmt  = new format(highlight(format("\u2026")));
     g_ellipsis_fmt    = new format(highlight(format("...")));
     g_placeholder_fmt = new format(highlight(format("_")));
-    g_lambda_n_fmt    = new format(highlight_keyword(format("\u03BB")));
+    g_lambda_n_fmt    = new format(highlight_keyword(format("\xCE\xBB"))); //\u03BB
     g_lambda_fmt      = new format(highlight_keyword(format("fun")));
```
Comment (leodemoura) [inline src/frontends/lean/pp.cpp]: @nunoplopes Can we put the unicode character (λ) instead? g++ and clang++ have support for it.
```
    g_lambda_n_fmt    = new format(highlight_keyword(format("λ")));		
```
BTW, if VS does not support it, we will have to fix other places where we have used unicode characters directly.

File: src/frontends/lean/pp.cpp [hunk 2]
```diff
@@ -127,13 +127,13 @@ void initialize_pp() {
...
-    g_forall_n_fmt    = new format(highlight_keyword(format("\u2200")));
+    g_forall_n_fmt    = new format(highlight_keyword(format("\xE2\x88\x80"))); //\u2200
     g_forall_fmt      = new format(highlight_keyword(format("forall")));
     g_pi_n_fmt        = new format(highlight_keyword(format("Π")));
```
Comment (leodemoura) [inline src/frontends/lean/pp.cpp]: Here is a place where we used the unicode character directly.

Comment (nunoplopes) [reply src/frontends/lean/pp.cpp]: it works fine. I've updated the patch.

**Comment by leodemoura** [comment]
merged

**Pull request closed without merging**